### DISCUSSION TITLE
能不能什么方案可以避免开发时每次更新会把变量示例清除？

### DISCUSSION THREAD

**Discussion by xmsz** [midwayjs/midway#2232] [General / 闲聊]
背景
- 我有个变量是储存一些class实例
- 然后我现在每次更改代码刷新后，这个变量也会重置

需求
- 有没有什么方案可以即使代码刷新服务重载，但是某个变量或者实例可以继续存在。
- 比如持久化变量/状态或者某个文件一直存在内存不刷新，或者https://stackoverflow.com/questions/46045535/persistence-of-a-variable-through-program-restarts

场景

储存多个第三方服务的长连接实例，现在每次开发，都要重连，但是重连会导致一些其他问题很麻烦

**Comment by flyingcrp** [comment]
如果能解决这个问题对于第三方依赖较多的情况下应该会很大的提升更改代码后的重启速度. 
集成测试的时候也会快很多

**Reply by czy88840616** [reply]
node 在服务端开发下至今为此就没有 HMR 相关的技术。不过如果只是本地开发，并且不考虑内存泄露，以及三方包的场景，理论上是可行的，比如把容器对象换掉就好了。
